### PR TITLE
golangci-lint fix ineffassign

### PR DESCRIPTION
**Pull request by MinerYang**
Signed-off-by: yminer <yminer@vmware.com>

Thank you for contributing to Harbor!

# Comprehensive Summary of your change

# Issue being fixed
Fixes #(issue)

Please indicate you've done the following:
- [ ] Well Written Title and Summary of the PR
- [ ] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [ ] Accepted the DCO. Commits without the DCO will delay acceptance.
- [ ] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/lib/redis/helper.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/goharbor/harbor/src/lib/log"
```
Comment (wy65701436) [inline src/lib/redis/helper.go:23]: seems that the helper.go can be removed?

File: src/pkg/replication/dao/dao.go [hunk 2]
```diff
@@ -104,6 +104,9 @@ func (d *dao) Update(ctx context.Context, policy *model.Policy, props ...string)
 	if e := orm.AsConflictError(err, "replication policy %s already exists", policy.Name); e != nil {
```
Comment (wy65701436) [inline src/pkg/replication/dao/dao.go:104]:
```
       if err != nil {
		if e := orm.AsConflictError(err, "replication policy %s already exists", policy.Name)); e != nil {
			err = e
		}
		return err
	}
```

**Review by wy65701436 — APPROVED** [review]
lgtm

File: src/server/registry/catalog.go [hunk 3]
```diff
@@ -75,7 +75,7 @@ func (r *repositoryHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 	}
 
 	// handle the pagination
-	resRepos := repoNames
+	var resRepos []string
```
Comment (zyyw) [inline src/server/registry/catalog.go:78]: assuming `repoNames` is not empty
```
resRepos := repoNames
```
is not equivalent with
```
var resRepos []string
```

Comment (MinerYang) [reply src/server/registry/catalog.go:78]: It would be assigned value, so it is a ineffective assign here indeed.

**Review by zyyw — APPROVED** [review]

**Pull request merged**